src/v2/cookbook/client-side-storage.md

```diff
@@ -0,0 +1,162 @@
+<script async src="https://static.codepen.io/assets/embed/ei.js"></script>
+
+## Working with Complex Values
+
+As mentioned above, Local Storage only works with simple values. To store more complex values, like arrays, you must serialize and deserialize the values with JSON. Here is a more advanced example that persists an array of cats (the best kind of array possible). 
+
+``` html
+<div id="app">
+  <h2>Cats</h2>
+  <div v-for="(cat,n) in cats">
+    <p>
+    <span class="cat">{{cat}}</span> <button @click="removeCat(n)">Remove</button>
+    </p>
+  </div>
+  
+  <p>
+    <input v-model="newCat"> 
+    <button @click="addCat">Add Cat</button>
+  </p>
+  
+</div>
+```
+
+This "app" consists of a simple list on top (with a button to remove a cat) and a small form at the bottom to add a new cat. Now let's look at the JavaScript.
+
+``` js
+const app = new Vue({
+  el:'#app',
+  data:{
+    cats:[],
+    newCat:null
+  },
+  mounted() {
+    
+    if(localStorage.getItem('cats')) {
+      try {
+        this.cats = JSON.parse(localStorage.getItem('cats'));
+      } catch(e) {
+        localStorage.removeItem('cats');
+      }
+    }
+  },
+  methods:{
+    addCat() {
+      // ensure they actually typed something
+      if(!this.newCat) return;
+      this.cats.push(this.newCat);
+      this.newCat = '';
+      this.saveCats();
+    },
+    removeCat(x) {
+      this.cats.splice(x,1);
+      this.saveCats();
+    },
+    saveCats() {
+      let parsed = JSON.stringify(this.cats);
+      localStorage.setItem('cats', parsed);
+    }
+  }
+})
+```
+
+In this application, we've switched to use the Local Storage APIs versus "direct" access. Both work but the API method is generally preferred. `mounted` now has to grab the value and parse the JSON value. If anything goes wrong here we assume the data is corrupt and delete it. (Remember, any time your web application uses client-side storage, the user has access to it and can modify it at will.)
+
+We have three methods now to handle working with cat. Both `addCat` and `removeCat` handle updating the "live" Vue data stored in `this.cats`. They then run `saveCats` which handles serializing and persisting the data. You can play with this version below:
+
+<p data-height="265" data-theme-id="0" data-slug-hash="qoYbyW" data-default-tab="js,result" data-user="cfjedimaster" data-embed-version="2" data-pen-title="localstorage, complex" class="codepen">See the Pen <a href="https://codepen.io/cfjedimaster/pen/qoYbyW/">localstorage, complex</a> by Raymond Camden (<a href="https://codepen.io/cfjedimaster">@cfjedimaster</a>) on <a href="https://codepen.io">CodePen</a>.</p>
+<script async src="https://static.codepen.io/assets/embed/ei.js"></script>
+
+## Alternative Patterns

```

Comment:
I like this alternative patterns section but it might be nice to have a wrapping up section.

Reply:
Query - would that go before or after the Alternative Patterns?

Reply:
"Wrapping up" is just a sentence or two at the end, after the Alternative Patterns section